### ISSUE TITLE
[BUG] trace-agent: a time.Timer leak in HttpReceiver.handleTraces

### ISSUE THREAD

**Issue by deadok22** [DataDog/datadog-agent#31468]
Upon a successful semaphore acquisition, the resources associated with the `time.After` call are left behind [here](https://github.com/DataDog/datadog-agent/blob/924a1502fdb70f35064838fa1bbf27b54459fb1b/pkg/trace/api/api.go#L573-L574). 

This code is run once on every spans-accepting endpoint call, so up to `apm_config.decoder_timeout` (default: 1 second) times the endpoint-call-rate active `time.Timer` instances are retained. That leads to inflated heap data structures backing `time.Timer` (slower addition of new timers) and CPU time + context switches wasted on the timer firing logic.

Stopping the timer before continuing is the way to go. E.g. like so:
```
	decoderTimeout := time.NewTimer(time.Duration(r.conf.DecoderTimeout) * time.Millisecond)
	select {
	case r.recvsem <- struct{}{}:
		_ = decoderTimeout.Stop()
	case <-decoderTimeout.C:
        ...
        }
```

**Comment by deadok22** [comment]
I opened https://help.datadoghq.com/hc/en-us/requests/1969667 to get attention to this ticket.

**Comment by ajgajg1134** [comment]
Hello! Thanks for opening an issue here, this does seem like an area where we might be able to make a small improvement. Before doing that I'll need to implement a benchmark to be sure of the performance improvement as I have never seen these timers show up in any of our internal heap or cpu profiles on very busy trace-agents. 🤔  Did you discover this change from seeing issues in your trace agent and this area appeared in your profiles?

**Comment by deadok22** [comment]
Hey @ajgajg1134,

I spotted the issue statically while reading the code to troubleshoot something unrelated.

When this grabbed my attention I wondered if anything would be affected and I found this change by @ahmed-mez (https://github.com/DataDog/datadog-agent/pull/18868) - increasing `DecoderTimeout` may well have resulted in no timers firing until after all tests complete, which "fixed" flakiness :)

Starting with Go 1.23, this is a non-issue. Calling Stop as I suggested above is not needed because of this awesome new feature: https://github.com/golang/go/commit/508bb17edd04479622fad263cd702deac1c49157

**Comment by ajgajg1134** [comment]
👍 That's good news then since the agent is currently built / released with 1.23 https://github.com/DataDog/datadog-agent/blob/main/go.mod#L3 

I also ran a quick benchmark internally with this change and there wasn't a measurable difference in cpu / memory compared to  the pre-release version of 7.61.

Given that, I'm going to mark this issue as closed. Thanks for pointing this out regardless! If you have additional comments or data feel free to re-open or bring it up in your support ticket.
